### DISCUSSION TITLE
Does Avalonia include a "Picking" function where one can detect all elements (in order top to bottom) in the hierarchy at a given screen position?

### DISCUSSION THREAD

**Discussion by jonmdev** [AvaloniaUI/Avalonia#14844] [Q&A] [answered]
I am getting very eager to move to Avalonia but after wasting time in other systems that ended up not having what I need, I am hopeful to clarify first.

I need to manage my own user clicking behaviors as much as possible. Does Avalonia feature a "Picking" type function that allows one to specify a screen position and it returns the elements that are in that spot in order from top to bottom?

Do we also have access to the click start/position/drag/end data (for example by events) as each click occurs?

Thanks for any clarification or documentation on these functions and what they are called.

**Comment by maxkatz6** [comment, marked as answer]
> Does Avalonia feature a "Picking" type function that allows one to specify a screen position and it returns the elements that are in that spot in order from top to bottom?

Do you mean, Hit Test? Each control and window have `InputHitTest` method that returns visual under the point specified in the args. If you need many visuals that were matched by that point - `GetInputElementsAt` or `GetVisualsAt` (last one is a bit broader, as might return non-interactable visuals as well).

> Do we also have access to the click start/position/drag/end data (for example by events) as each click occurs?

Each pointer event argument keeps current point information like position, pressure, pressed buttons. Combining these events (PointerPressed + PointerMoved + PointerReleased or Cancelled), you can calculate position delta if you need that.
